internal/command/launch/launch.go

```diff
@@ -210,17 +200,43 @@ func run(ctx context.Context) (err error) {
 		}
 	}
 
-	if !flag.GetBool(ctx, "generate-name") {
-		if appName := flag.GetString(ctx, "name"); appName == "" {
-			// Prompt the user for the app name
-			if inputName, err := prompt.SelectAppName(ctx); err != nil {
+	if generateName {
+		appConfig.AppName = ""
+	}
+
+	if name != "" {
+		appConfig.AppName = name
+	}
+
+	if !generateName && name == "" {
+		inputName, err := promptForAppName(ctx, cfg)
+		if err != nil {
+			return err
+		}
+		appConfig.AppName = inputName
+	}
+
+	if appConfig.AppName != "" {
+		running, err := isAppRunning(ctx, appConfig)
+		if err != nil {
+			return err
+		}
+
+		if running {
+			msg := fmt.Sprintf("%s is already running, do you want to still want to deploy into that app?", appConfig.AppName)
+			yes, err := prompt.Confirm(ctx, msg)
+			if err != nil {
 				return err
-			} else {
-				appConfig.AppName = inputName
 			}
-		} else {
-			appConfig.AppName = appName
+
+			if yes {
+				ctx = appconfig.WithName(ctx, cfg.AppName)
+				return deploy.DeployWithConfig(ctx, cfg, deployArgs)
```

Comment:
The whole point of https://github.com/superfly/flyctl/pull/1872#discussion_r1131418836 was to remove this "shortcut to deploy".

`--yes` will continue with normal `fly launch` even if the app exists, see https://github.com/superfly/flyctl/pull/1872#discussion_r1131656793. `fly launch` will offer the user to deploy at the end of the flow as it normally does.